Crash: AssertionError "asTerm called on not-a-Term type T" when exporting members of a cyclic opaque type companion

## Compiler version

- 3.8.3 — crashes
- 3.7.4 — crashes
- 3.6.4 — crashes
- 3.5.2 — crashes

## Minimized code

```scala
//> using scala 3.8.3

object core:
  opaque type T = String | List[T]
  object T:
    def make(s: String): T = s

object api:
  export core.T
  export core.T.*
```

Compile with `scala-cli compile --server=false repro.scala`.

## Output (click arrow to expand)

<details>

```
-- [E140] Cyclic Error: repro.scala:4:14 ------------------------------------
4 |  opaque type T = String | List[T]
  |              ^
  |illegal cyclic type reference: alias String | List[core.T] of type T refers back to the type itself

Exception in thread "main" java.lang.AssertionError: assertion failed: asTerm called on not-a-Term type T
  at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
  at dotty.tools.dotc.core.Symbols$Symbol.asTerm(Symbols.scala:186)
  at dotty.tools.dotc.typer.Namer.dotty$tools$dotc$typer$Namer$ClassCompleter$$_$_$$anonfun$19(Namer.scala:1371)
  at dotty.tools.dotc.ast.tpd$.DefDef(tpd.scala:316)
  at dotty.tools.dotc.typer.Namer$ClassCompleter.addForwarder$1(Namer.scala:1383)
  at dotty.tools.dotc.typer.Namer$ClassCompleter.addForwardersNamed$1$$anonfun$1(Namer.scala:1404)
  ...
  at dotty.tools.dotc.typer.Namer$ClassCompleter.exportForwarders(Namer.scala:1511)
  at dotty.tools.dotc.typer.Namer$ClassCompleter.processExport$1(Namer.scala:1522)
```

</details>

## What I expected

The cyclic-type error to be reported and the compiler to exit cleanly, the same way it does when the two `export` lines are removed.

## Notes

- Without the `export` lines, the cyclic error is reported and the compiler exits cleanly.
- Replacing `opaque type T` with a plain `type T = String | List[T]` reports a different (also cyclic) error and does not crash — the `opaque` qualifier is essential because it creates a companion object whose members `export core.T.*` then tries to forward.
- Both lines are needed; either alone also crashes in 3.8.3.

The crash happens in `exportForwarders` (https://github.com/scala/scala3/blob/c3e3e42f20c02ee9cdeaa0ae1420555eb4ea978d/compiler/src/dotty/tools/dotc/typer/Namer.scala#L1511) when the Namer reaches a member of `core.T` that is the cyclic type alias `T` itself, and calls `asTerm` on it (https://github.com/scala/scala3/blob/c3e3e42f20c02ee9cdeaa0ae1420555eb4ea978d/compiler/src/dotty/tools/dotc/core/Symbols.scala#L186). It looks like the export-forwarder code should check for a type/error symbol before calling `asTerm`, or skip forwarding when the source companion contains a symbol that hasn't completed cleanly.

Existing related issues in the same neighbourhood (different triggers, separate stacks):
- #25702 — same `asTerm` assertion, but needs cats/fs2 deps and an incomplete class
- #24014, #24051 — `export` interactions with opaque types, no crash

I'm fairly sure this is a distinct one. Happy to bisect if helpful.

--- a/compiler/src/dotty/tools/dotc/typer/Namer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Namer.scala
@@ -1264,7 +1264,9 @@ class Namer { typer: Typer =>
                it suffices to check if symbol is the same class. */
             cls == id.symbol
           case _ => false
-        if !sym.isAccessibleFrom(pathType) then
+        if mbr.info.isInstanceOf[ErrorType] then
+          No("already has an error")
+        else if !sym.isAccessibleFrom(pathType) then
           No("is not accessible")
         else if sym.isConstructor || sym.is(ModuleClass) || sym.is(Bridge) || sym.is(PhantomSymbol) || sym.isAllOf(JavaModule) then
           Skip

--- a/compiler/test/dotc/neg-best-effort-pickling.excludelist
+++ b/compiler/test/dotc/neg-best-effort-pickling.excludelist
@@ -20,6 +20,7 @@ i11226.scala # missing type
 i974.scala # cyclic reference
 i13864.scala # missing symbol in pickling
 type-params.scala # recursion limit exceeded
+26037.scala # recursion limit exceeded
 
 unsafe-dollar-in-name.scala # missing owner in unpickler
 unsafe-dollar-in-name-2.scala # missing owner in unpickler

--- a/compiler/test/dotty/tools/dotc/Playground.scala
+++ b/compiler/test/dotty/tools/dotc/Playground.scala
@@ -10,8 +10,8 @@ import org.junit.Ignore
   import CompilationTests.{*, given}
   import CompilationTest.aggregateTests
 
-  @Test def example: Unit =
-    implicit val testGroup: TestGroup = TestGroup("single-test")
+  @Test def example(): Unit =
+    given testGroup: TestGroup = TestGroup("single-test")
     // can add, e.g., .and("-some-option")
     val options = defaultOptions
     // can also use `compileDir` (single test as a dir), `compileFilesInDir` (all tests within a dir)
@@ -21,3 +21,14 @@ import org.junit.Ignore
     val compilationTest = withCoverage(aggregateTests(test))
     runWithCoverageOrFallback[TestKind](compilationTest, testGroup.name)
 
+  @Test def bestEffortTasty(): Unit =
+    given testGroup: TestGroup = TestGroup("single-test")
+    // can add, e.g., .and("-some-option")
+    val options = TestConfiguration.bestEffortBaselineOptions
+    // make sure we only test one thing at a time, despite the framework requiring an entire dir + filters
+    val testFile = "26037.scala"
+    dotc.comptest.compileBestEffortTastyInDir(
+      "tests/neg", options,
+      picklingFilter = FileFilter.include(List(testFile)),
+      unpicklingFilter = FileFilter.include(List(testFile))
+    ).checkNoCrash()

--- a/compiler/test/dotty/tools/vulpix/ParallelTesting.scala
+++ b/compiler/test/dotty/tools/vulpix/ParallelTesting.scala
@@ -1684,7 +1684,7 @@ trait ParallelTesting extends RunnerOrchestration with CoverageSupport:
     assert(!flags.options.contains(bestEffortFlag), "Best effort compilation flag should not be added manually")
 
     val outDir = new JFile(defaultOutputDir, testGroup.name)
-    val sourceDir = new JFile(f)
+    val sourceDir = TestSources.getPath(f).toFile
     checkRequirements(f, sourceDir, outDir)
 
     val (dirsStep1, filteredPicklingFiles) = compilationTargets(sourceDir, picklingFilter)

--- a/tests/neg/26037.scala
+++ b/tests/neg/26037.scala
@@ -1,0 +1,8 @@
+object core:
+  opaque type T = String | List[T] // error: illegal cyclic type
+  object T:
+    def make(s: String): T = s
+
+object api:
+  export core.T
+  export core.T.*

